packages/graph/hash_graph/lib/graph/src/ontology/mod.rs

```diff
@@ -134,6 +136,26 @@ pub struct PersistedDataType {
     pub identifier: PersistedOntologyIdentifier,
 }
 
+/// Query to read [`DataType`]s, which are matching the [`Expression`].
+#[derive(Debug, Deserialize, Component)]
+#[serde(deny_unknown_fields, rename_all = "camelCase")]
+pub struct DataTypeQuery {
+    #[serde(rename = "query")]
+    pub expression: Expression,
+    /// Depth used to read nested [`DataType`]s.
```

Comment:
I'm still not a huge fan of us calling these nested, as that somewhat implies that they're defined within another one. They're just referenced.

I think we can (and should) be a little more precise about what depth means, not just for the sake of correctness but because I think it could be somewhat confusing. Especially considering each type has its own depth which is independent of the other depths, which means it's not quite the normal definition as people would expect in a graph query

Reply:
I think the doc-comment above on data_type_query_depth is pretty good by the way

Reply:
Maybe we should consider creating a `type Depth = u8` alias. We can then put documentation on the depth once (rather than copy pasting these all over the place) and export the type. It also makes it easier for us to allow deeper depths if we realise 255 is too small.